task12/CarsLibrary/CarsLibrary/Cars.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CarsLibrary
+{
+    public class Cars
+    {
+        public enum BodyType
+        {
+            Sedan,
+            Hatchback,
+            StationWagon,
+            Cabriolet
+        }
+
+        public class Car
+        {
+            public readonly string VIN;
+
+            public string Brand { get; set; }
+
+            public int YearOfRelease { get; set; }
+
+            public BodyType BodyType { get; set; }
+
+            public decimal Price { get; set; }
+
+            public DateTime DateOfSale { get; set; }
+
+            public string CustomerFullName { get; set; }
+
+            public Car(string brand, int yearOfRelease, string vin, BodyType bodyType)
+            {
+                Brand = brand;
+                if (yearOfRelease <= 0)
+                    throw new ArgumentException();
+                YearOfRelease = yearOfRelease;
+
+                if (vin.Length != 17)
+                    throw new ArgumentException();
+                VIN = vin;
+                BodyType = bodyType;
+            }
+
+            public string GetInfo()
+            {
+                var body = "";
+                if (BodyType == BodyType.Sedan)
+                    body = "Sedan";
+                else if (BodyType == BodyType.Hatchback)
+                    body = "Hatchback";
+                else if (BodyType == BodyType.StationWagon)
+                    body = "StationWagon";
+                else if (BodyType == BodyType.Cabriolet)
+                    body = "Cabriolet";
```

Comment:
Можно проще, без каскада ветвлений `if-else`
```suggestion
                var body = BodyType.ToString();
```